include/gsl/span

```diff
@@ -262,22 +289,22 @@ namespace details
         // algorithm calls
         friend constexpr void _Verify_range(span_iterator lhs, span_iterator rhs) noexcept
         { // test that [lhs, rhs) forms a valid range inside an STL algorithm
-            Expects(lhs.span_ == rhs.span_        // range spans have to match
-                    && lhs.index_ <= rhs.index_); // range must not be transposed
+            Expects(lhs.begin_ == rhs.begin_    // range spans have to match
+                    && lhs.end_ <= rhs.end_); // range must not be transposed
         }
 
         constexpr void _Verify_offset(const difference_type n) const noexcept
         { // test that the iterator *this + n is a valid range in an STL
             // algorithm call
-            Expects((index_ + n) >= 0 && (index_ + n) <= span_->size());
+            Expects((current_ + n) >= begin_ && (current_ + n) <= end_);
```

Comment:
This Expects has been written to trigger pointer overflow and then look at the consequences, which is potentially problematic. I recommend directly testing the unknown `n` against known-valid expressions like `end_ - current_`.